Create system events reference section (#787)
inngest/website: pages/docs/reference/system-events/inngest-function-failed.mdx

# `inngest/function.failed` {{ className: "not-prose" }}

> The `inngest/function.failed` event is triggered whenever any single function fails in your Inngest [environment]((/docs/platform/environments)).

Reviewer: ```suggestion
The `inngest/function.failed` event is sent whenever any single function fails in your Inngest [environment]((/docs/platform/environments)).
```

> This event can be used to track all function failures in a single place, enabling you to send metrics, alerts, or events to external systems like Datadog or Sentry for all of your Inngest functions.

Reviewer 2: ```suggestion
This event can be used to track all function failures in a single place, enabling you to send metrics, alerts, or events to [external systems like Datadog or Sentry](/docs/examples/track-failures-in-datadog) for all of your Inngest functions.
```

Do you think it makes sense to link this page here?

> The TypeScript SDK offers a short hand `onFailure` handler that can be used to handle this event for a specific function.

Reviewer: Python SDK has an `on_failure` handler. Should we have some sort of language support list for features? Various features have varying SDK support

Reviewer 2: ```suggestion
The TypeScript SDK offers a shorthand `onFailure` handler that can be used to handle this event for a specific function.
```

Author: I'll make this more generic like "our sdks offer short hand "on failure" handlers to that can be used to handle this for a specific function then link to both in the end of the page.

## The event payload

<Row>
<Col>

<Properties>
  <Property name="name" type={`string: "inngest/function.failed"`} attribute>
    The `inngest/` event prefix is reserved for system events in each environment.
  </Property>
  <Property name="data" type="object" attribute>
    The event payload data.
    <Properties nested>
      <Property name="error" type="object" attribute>
        Data about the error payload as returned from the failed function.
        <Properties nested>
          <Property name="message" type="string" attribute>

>             The error message when an error is caught

Reviewer 2: ```suggestion
            The error message when an error is caught.
```

          </Property>
          <Property name="name" type="string" attribute>

>             The name of the error, defaulting to "Error" if unspecified

Reviewer 2: ```suggestion
            The name of the error, defaulting to "Error" if unspecified.
```

          </Property>
          <Property name="stack" type="string">

>             The stack trace of the error, if supported by the language SDK

Reviewer 2: ```suggestion
            The stack trace of the error, if supported by the language SDK.
```

          </Property>
        </Properties>
      </Property>
      <Property name="event" type="string" attribute>
        The failed function's original event payload.
      </Property>
      <Property name="function_id" type="string" attribute>

>         The failed function's [`id`](/docs/reference/functions/create#configuration)

Reviewer 2: ```suggestion
        The failed function's [`id`](/docs/reference/functions/create#configuration).
```

      </Property>
      <Property name="run_id" type="string" attribute>

>         The failed function's function run ID.

Reviewer 2: ```suggestion
        The failed function's [run ID](/docs/reference/functions/create#run-id).
```

Author: This makes me think a language-neutral reference of things like run ids and stuff could be interesting as well 🤔
